tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs

```diff
@@ -126,6 +123,95 @@ public async Task ConfidentialClientWithClientSecretTestAsync()
+                .Build();
+
+            var authResult = await confidentialApp.AcquireTokenForClient(s_keyvaultScope)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            MsalAssert.AssertAuthResult(authResult);
+        }
+
+        [TestMethod]
+        public async Task ConfidentialClientWithSignedAssertionTestAsync()
+        {
+            var keyvault = new KeyVaultSecretsProvider();
+            var secret = keyvault.GetSecret(MsalTestConstants.MsalCCAKeyVaultUri).Value;
+            var confidentialClientAuthority = "https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47";
+
+            X509Certificate2 cert = GetCertificate();
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                .Create(ConfidentialClientID)
+                .WithAuthority(new Uri(confidentialClientAuthority), true)
+                .WithClientAssertion(GetSignedClientAssertion(ConfidentialClientID))
+                .Build();
+
+            var authResult = await confidentialApp.AcquireTokenForClient(s_keyvaultScope)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            MsalAssert.AssertAuthResult(authResult);
+        }
+
+        internal static long ConvertToTimeT(DateTime time)
+        {
+            var startTime = new DateTime(1970, 1, 1, 0, 0, 0, 0);
+            TimeSpan diff = time - startTime;
+            return (long)diff.TotalSeconds;
+        }
+
+        private static IDictionary<string, string> GetClaims()
+        {
+            DateTime validFrom = DateTime.UtcNow;
+            var nbf = ConvertToTimeT(validFrom);
+            var exp = ConvertToTimeT(validFrom + TimeSpan.FromSeconds(MsalTestConstants.JwtToAadLifetimeInSeconds));
+
+            return new Dictionary<string, string>()
+            {
+                { "aud", MsalTestConstants.ClientCredentialAudience },
+                { "exp", exp.ToString(CultureInfo.InvariantCulture) },
+                { "iss", ConfidentialClientID.ToString(CultureInfo.InvariantCulture) },
+                { "jti", Guid.NewGuid().ToString() },
+                { "nbf", nbf.ToString(CultureInfo.InvariantCulture) },
+                { "sub", ConfidentialClientID.ToString(CultureInfo.InvariantCulture) }
+            };
+        }
+
+        private static string GetSignedClientAssertion(string clientId)
+        {
+            var manager = new NetDesktopCryptographyManager();
+            var jwtToken = new JsonWebToken(manager, clientId, MsalTestConstants.ClientCredentialAudience);
+
+            var clientCredential = ClientCredentialWrapper.CreateWithCertificate(GetCertificate());
+            return jwtToken.Sign(clientCredential, false);
+        }
+
+        private static X509Certificate2 GetCertificate()
+        {
+            X509Certificate2 cert = CertificateHelper.FindCertificateByThumbprint("79fbcbeb5cd28994e50daff8035bacf764b14306");
```

Comment:
>"79fbcbeb5cd28994e50daff8035bacf764b14306" [](start = 82, length = 42)

named const? Is this a real cert? Where does it exist? How will the dev machine get this?

Reply:
This is the same cert we use for all of our automation. it is available to download in our keyvault. Ill add a named const for the thumbprint.